src/k8s/cmd/k8sd/k8sd.go

```diff
@@ -71,6 +73,7 @@ func NewRootCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 	cmd.PersistentFlags().BoolVar(&rootCmdOpts.disableUpdateNodeConfigController, "disable-update-node-config-controller", false, "Disable the Update Node Config Controller")
 	cmd.PersistentFlags().BoolVar(&rootCmdOpts.disableFeatureController, "disable-feature-controller", false, "Disable the Feature Controller")
 	cmd.PersistentFlags().BoolVar(&rootCmdOpts.disableCSRSigningController, "disable-csrsigning-controller", false, "Disable the CSR signing controller")
+	cmd.PersistentFlags().IntVar(&rootCmdOpts.maxPreRemoveHookDBTxnRetries, "max-remove-txn-retries", 10, "Maximum number of retries for database transactions in the pre-remove hook")
```

Comment:
This doesn't seem very useful to have as an exposed flag, and exposes too much implementation detail imo. I think the timeout flag covers the vast majority of use cases.

Reply:
so do you suggest marking this a hidden flag with `cmd.PersistentFlags().MarkHidden("max-remove-txn-retries")` or completely remove the flag and hard code the num retries?